src/popup/App.vue

```diff
@@ -274,42 +274,49 @@ export default {
 
       return newDescription;
     },
+    logCanSync (log) {
+      return log.issue !== 'NO ID' && !log.isSynced && log.duration >= 60;
+    },
     async syncToJira () {
       const _self = this;
+      _self.isSyncing = true;
       const headers = {
         'X-Atlassian-Token': 'no-check', 'User-Agent': ''
       };
       const awaitingIssues = {};
       for (let log of this.checkedLogs) {
-        if (log.issue in awaitingIssues) {
-          await awaitingIssues[log.issue];
-        }
-        const promise = axios({
-          method: 'post',
-          url:
-            _self.jiraUrl + '/rest/api/latest/issue/' + log.issue + '/worklog',
-          data: {
-            timeSpentSeconds: log.duration,
-            comment: log.description,
-            started: _self.toJiraDateTime(log.start)
-          },
-          headers: headers
-        })
-          .then(function (response) {
-            _self.isSaving = false;
-            _self.showSnackbar = true;
-            _self.checkIfAlreadyLogged(log);
-            _self.checkedLogs = [];
-            _self.syncAllLogs = false;
-          })
-          .catch(function (error) {
-            _self.errorMessage = error;
+        if (_self.logCanSync(log)) {
+          if (log.issue in awaitingIssues) {
+            await awaitingIssues[log.issue];
+          }
+          const promise = axios({
+            method: 'post',
+            url:
+              _self.jiraUrl + '/rest/api/latest/issue/' + log.issue + '/worklog',
+            data: {
+              timeSpentSeconds: log.duration,
+              comment: log.description,
+              started: _self.toJiraDateTime(log.start)
+            },
+            headers: headers
           })
-          .finally(function () {
-            delete awaitingIssues[log.issue];
-          });
-        awaitingIssues[log.issue] = promise;
+            .then(async function (response) {
+              await _self.checkIfAlreadyLogged(log)
+            })
+            .catch(function (error) {
+              _self.errorMessage = error;
+            })
+            .finally(function () {
+              delete awaitingIssues[log.issue];
+            });
+          awaitingIssues[log.issue] = promise;
+        }
       }
+      await Promise.all(Object.values(awaitingIssues))
+      _self.isSyncing = false;
+      _self.showSnackbar = true;
+      _self.checkedLogs = [];
+      _self.syncAllLogs = false;
```

Comment:
Just curious for my learning about the reason for moving these out of the `then`?

Reply:
They could technically go in either place since this code will only run after all `then` are executed. Moving them out will just stop these being reassigned the same values for each promise that resolves. `_self.checkIfAlreadyLogged` is the only one that needed to stay as that is called per `log`.